### PR TITLE
feat(Select): Convert `Select` component to CSS Modules behind feature flag

### PR DESCRIPTION
**Pull request by randall-krauskopf**
Closes https://github.com/github/primer/issues/4299

#### Changed
* Update the base primer exports and other internal references to use the Select component used in `src/Select/Select.tsx` instead of the duplicate one located at `src/Select.tsx` which was causing some confusion. 
* Convert select component to css module behind feature flag.

### Rollout strategy

- [ ] Patch release
- [x] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### CODE REVIEW

File: packages/react/src/FormControl/FormControl.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import Autocomplete from '../Autocomplete'
 import Box from '../Box'
 import Checkbox from '../Checkbox'
 import Radio from '../Radio'
-import Select from '../Select'
+import Select from '../Select/Select'
```
Comment (randall-krauskopf) [inline packages/react/src/FormControl/FormControl.tsx:6]: Use the newest `Select` component

File: packages/react/src/Select/Select.features.stories.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import React from 'react'
-import {Select, FormControl, Box, Heading} from '..'
+import {FormControl, Box, Heading} from '..'
+import Select from '.'
```
Comment (randall-krauskopf) [inline packages/react/src/Select/Select.features.stories.tsx:3]: Get `Select` from the correct location

File: packages/react/src/Select/Select.figma.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import React from 'react'
-import {Select} from '../../src'
+import Select from '.'
```
Comment (randall-krauskopf) [inline packages/react/src/Select/Select.figma.tsx:2]: Get `Select` from the correct location

File: packages/react/src/index.ts [hunk 4]
```diff
@@ -137,8 +137,9 @@ export {default as RadioGroup} from './RadioGroup'
 export type {RelativeTimeProps} from './RelativeTime'
 export {default as RelativeTime} from './RelativeTime'
 export {SegmentedControl} from './SegmentedControl'
-export {default as Select} from './Select'
-export type {SelectProps} from './Select'
+// Curently there is a duplicate Select component at the root of the dir, so need to be explicit about exporting from the src/Select dir
+export {default as Select} from './Select/Select'
```
Comment (randall-krauskopf) [inline packages/react/src/index.ts:141]: @joshblack Should I explore deleting the `src/Select.tsx` file in a follow up PR?

Comment (joshblack) [reply packages/react/src/index.ts:141]: That would be very much so appreciated if you have the time! 🙏

**Review by jonrohan — APPROVED** [review]

**Pull request merged**